official/boosted_trees_upgraded/train_higgs_test.py

```diff
@@ -0,0 +1,157 @@
+# Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ==============================================================================
+"""Tests for boosted_tree."""
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tempfile
+
+import numpy as np
+import pandas as pd
+import tensorflow as tf
+
+from absl import logging
+# pylint: disable=g-bad-import-order
+from official.boosted_trees_upgraded import train_higgs
+from official.utils.testing import integration
+
+TEST_CSV = os.path.join(os.path.dirname(__file__), "train_higgs_test.csv")
+
+logging.set_verbosity(logging.ERROR)
+
+
+class BaseTest(tf.test.TestCase):
+  """Tests for Wide Deep model."""
+
+  @classmethod
+  def setUpClass(cls):  # pylint: disable=invalid-name
+    super(BaseTest, cls).setUpClass()
+    train_higgs.define_train_higgs_flags()
+
+  def setUp(self):
+    # Create temporary CSV file
+    self.data_dir = self.get_temp_dir()
+    data = pd.read_csv(
+        TEST_CSV, dtype=np.float32, names=["c%02d" % i for i in range(29)]
+    ).as_matrix()
+    self.input_npz = os.path.join(self.data_dir, train_higgs.NPZ_FILE)
+    # numpy.savez doesn't take gfile.Gfile, so need to write down and copy.
+    tmpfile = tempfile.NamedTemporaryFile()
+    np.savez_compressed(tmpfile, data=data)
+    tf.io.gfile.copy(tmpfile.name, self.input_npz)
+
+  def test_read_higgs_data(self):
+    """Tests read_higgs_data() function."""
+    # Error when a wrong data_dir is given.
+    with self.assertRaisesRegexp(RuntimeError, "Error loading data.*"):
+      train_data, eval_data = train_higgs.read_higgs_data(
+          self.data_dir + "non-existing-path",
+          train_start=0, train_count=15, eval_start=15, eval_count=5)
+
+    # Loading fine with the correct data_dir.
+    train_data, eval_data = train_higgs.read_higgs_data(
+        self.data_dir,
+        train_start=0, train_count=15, eval_start=15, eval_count=5)
+    self.assertEqual((15, 29), train_data.shape)
+    self.assertEqual((5, 29), eval_data.shape)
+
+ @tf.function
```

Comment:
why is this needed?

Reply:
I used tf.function wrapper because original file contains sess.run calls.